content/docs/api-reference/dvcfs.md

```diff
@@ -0,0 +1,66 @@
+# DvcFileSystem
+
+DvcFileSystem provides a pythonic file interface (aka
+[fsspec](https://filesystem-spec.readthedocs.io/)) for a DVC repo and provides a
+read-only filesystem-like operations like `ls`, `du`, `glob`, `get`, `download`,
+etc.
+
+DvcFileSystem provides a single view of all the files/directories in your
+repository, be it Git-tracked or DVC-tracked, or untracked (in case of a local
+repository). DvcFileSystem is smart to reuse the files in your cache directory
+(if present) and can otherwise stream/fetch the files from your default remote.
+
+### Basic Usage
+
+```py
+>>> from dvc.fs.dvc import _DvcFileSystem as DvcFileSystem
+# opening a local repository
+>>> fs = DvcFileSystem("/path/to/local/repository")
+# opening a remote repository
+>>> remote_fs = DvcFileSystem("https://github.com/iterative/example-get-started.git", rev="main")
+```
+
+## Examples
+
+### Listing all DVC-tracked files
+
+```py
+>>> fs = DvcFileSystem("https://github.com/iterative/example-get-started.git", rev="main")
+>>> fs.find("/", detail=False, dvc_only=True)
+[
+    '/data/data.xml',
+    '/data/features/test.pkl',
+    '/data/features/train.pkl',
+    '/data/prepared/test.tsv',
+    '/data/prepared/train.tsv',
+    '/evaluation/importance.png',
+    '/model.pkl'
+]
+```
+
+### Downloading a file or a directory
+
+```py
+>>> fs = DvcFileSystem("https://github.com/iterative/example-get-started.git", rev="main")
+>>> fs.get_file("data/data.xml", "data.xml")
+```
+
+This downloads "data/data.xml" file to the current working directory as
+"data.xml" file. The DVC-tracked files may be downloaded from the cache if it
+exists or may get streamed from the remote.
+
+```py
+>>> fs = DvcFileSystem("https://github.com/iterative/example-get-started.git", rev="main")
+>>> fs.get("data", "data", recursive=True)
+```
+
+This downloads all the files in "data" directory - be it Git-tracked or
+DVC-tracked into a local directory "data". Similarly, DVC might fetch files from
+remote if they don't exist in the cache.
+
+## API Reference
+
+As DvcFileSystem is based on [fsspec](https://filesystem-spec.readthedocs.io/),
+it is compatible with most of the APIs that it offers. Please check the fsspec's
```

Comment:
```suggestion
it is compatible with most of the read-only methods that it offers. Please check the fsspec's
```

Reply:
It's not the API that is read-only, but the filesystem and it's operation in general. I have also myself struggled to explain this, but tbh I don't think we need to mention anything other than "... is a read-only filesystem".

Reply:
I think it's better to even skip it and raise a `EROFS`.

Reply:
What is an `EROFS`?

I must be missing your point here because I don't see what's confusing or ambiguous. Is dvcfs compatible with the read-only methods of https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem?

Reply:
EROFS is an errno which is raised for "Read Only FileSystem". What I am trying to say is that the filesystem in general is read-only, not the methods. Take `fs.open()` for example. It works with `mode="rb"` but `mode="wb"` raises `EROFS`. So I am trying to distinguish between the method `open` vs the operations: writing/reading, etc.